### DISCUSSION TITLE
Use the same session inside threadpool but with a Lock

### DISCUSSION THREAD

**Discussion by mat105** [sqlalchemy/sqlalchemy#10026] [Usage Questions]
Hi, i have a sync operation that's working fine, but it proved to be too slow when i needed to run many of them one after another, i'm planning on starting a threadpoolexecutor to run the tasks.
I think i've read that using the same session on many threads isn't safe, but what if i add a lock?
something like this

```
from threading import Lock

# called many times concurrently
def my_func(session):
	response = requests.get("") # <- Slow request
	entity = something(response)

	with Lock():
		session.add(entity)
		session.commit(entity)
```
That session is obtained from a scoped_session, will something like this work? even with sqlite?.

**Comment by CaselIT** [comment]
Hi,

If you are just adding an element and committing you could probably use multiple sessions, one for each thread.
Or you could just return the entity and add them at the end from the main thread.

Regarding the code you provided, it should be ok, but I would try to use one of the alternatives suggested above
